doc/Building.md

```diff
@@ -0,0 +1,199 @@
+```
+
+If you run into a gettext error while building glib, try `brew link --force gettext` to override the built-in Mac OS BSD gettext with GNU GetText.
+
+*Note:* in Mojave (10.14) you need to set a minimum deployment version higher than 10.04. Otherwise you'll get the error `ld: library not found for -lgcc_s.10.4`.
+```
+export MACOSX_DEPLOYMENT_TARGET=10.09
+```
+
+
+## Getting the source
+
+
+```
+git clone https://github.com/koreader/koreader.git
+cd koreader && ./kodev fetch-thirdparty
+```
+
+Building the emulator
+
+## Building and running the emulator
+
+To build an emulator on your current Linux or MacOS machine:
+
+```
+./kodev build
+```
+
+To run KOReader on your development machine:
+
+```
+./kodev run
+```
+
+
+You can specify the size and DPI of the emulator's screen using
+`-w=X` (width), `-h=X` (height), and `-d=X` (DPI).
+
+ There is also a convenience
+`-s` (simulate) flag with some presets like `kobo-aura-one`, `kindle3`, and
+`hidpi`. The latter is a fictional device with `--screen_width=1500`,
+`--screen_height=2000` and `--screen_dpi=600` to help ensure DPI scaling works correctly.
+
+Sample usage:
+
+```
+./kodev run -s=kobo-aura-one
+```
+
+To use your own koreader-base repo instead of the default one change the `KOR_BASE`
+environment variable:
+
+```
+make KOR_BASE=../koreader-base
+```
+
+This will be handy if you are developing `koreader-base` and you want to test your
+modifications with the KOReader frontend. NOTE: this only supports relative path for now.
+
+## Building for other platforms
+
+Once you have the emulator ready to rock you can [build for other platforms too](Building_targets.md).
+
+## Testing
+
+You may need to checkout the [circleci config file][circleci-conf] to setup up
+a proper testing environment. 
+
+Briefly, you need to install `luarocks` and then install `busted` with `luarocks`. The "eng" language data file for tesseract-ocr is also need to test OCR functionality. Finally, make sure that `luajit` in your system is at least of version 2.0.2.
+
+To automatically set up a number of primarily luarocks-related environment variables:
+
+```
+./kodev activate
+```
+
+To run unit tests:
+
+```
+./kodev test base
+./kodev test front
+```
+
+To run a specific unit test (for test development):
+
+```
+./kodev test front readerbookmark_spec.lua
+```
+
+NOTE: Extra dependencies for tests: `busted` and `ansicolors` from luarocks.
+
+To run Lua static analysis:
+
+```
+make static-check
+```
+
+NOTE: Extra dependencies for tests: `luacheck` from luarocks
+
+
+## Translations
+
+Please refer to [l10n's README][l10n-readme] to grab the latest translations
+from [the KOReader project on Transifex][koreader-transifex] with this command:
+
+```
+make po
+```
+
+If your language is not listed on the Transifex project, please don't hesitate
+to send a language request [here][koreader-transifex].
+
+### Variables in translation
+
+Some strings contain variables that should remain unaltered in translation.
```

Comment:
We should add a little extra warning. Not sure how much good it'll do, but it's happened. ;-)
```suggestion
Some strings contain variables that should remain unaltered in translation. These take the form of a `%` followed by a number from `1-99`, although you'll seldom see more than about 5 in practice. Please don't put any spaces between the `%` and its number. `%1` should always remain `%1`.
```